File: sdks/python/examples/market-data/historical_prices.py
import pmxt

def main():
    api = pmxt.Polymarket()
    events = api.fetch_events(query="Who will Trump nominate as Fed Chair?")
    event = events[0]

    market = api.filter_markets(event.markets, "Kevin Warsh")[0]

    # Use outcome.outcome_id for fetching historical data
    # (Polymarket: CLOB Token ID, Kalshi: Market Ticker)
    history = api.fetch_ohlcv(
        market.yes.outcome_id,
        resolution='1h',
        limit=5
    )
    print(history)

if __name__ == "__main__":
    main()

File: sdks/typescript/examples/market-data/historical_prices.ts
import pmxt from 'pmxtjs';

const main = async () => {
    const api = new pmxt.Polymarket();
    const events = await api.searchEvents('Who will Trump nominate as Fed Chair?');
    const event = events[0];

    // 2. Search for the specific Market within that event
    const market = event.searchMarkets('Kevin Warsh')[0];

    // Note: Use market.yes.id for the outcome ID (CLOB Token ID on Poly)
    const history = await api.fetchOHLCV(market.yes!.id, {
        resolution: '1h',
        limit: 5
    });

    console.log(history);
};

main();
